background/basicauth.js

```diff
@@ -38,11 +56,16 @@
       // Workaround for #112
       return {};
     }
-    return new BasicAuthDialog({
-      'url': details.url,
-      'host': details.challenger.host,
-      'realm': details.realm,
-      'page_host': pageHost
+    return getBlacklist().then((blacklist) => {
```

Comment:
I did this to fit in with the explicit promise already being used in the dialog, and because I'm not sure how transforming everything (even the early returns) would fit in with the API.

Reply:
Okay let's leave it then as is.